### ISSUE TITLE
Blockly.mainworkspace is being conflated with the Blockly.WorkspaceSvg

### ISSUE THREAD

**Issue by zfi** [parallaxinc/solo#377]
There are a number of places in the code base where there Blocky.mainworkspace object is being used instead of the Blocky.WokspaceSvg object. While this appears to work, it is not the correct use, based on Google Blockly's [documentation](https://developers.google.com/blockly/reference/js/Blockly#.inject) and [example](https://developers.google.com/blockly/guides/configure/web/fixed-size).
```
<script>
  var workspace = Blockly.inject('blocklyDiv',
      {toolbox: document.getElementById('toolbox')});
</script>
```
The workspace that is returned is the handle that should be used to reference the workspace. This is a different object than the Blocky.mainworkspace object. While it may appear to work, it is not guaranteed to do so.

I have replaced the workspace reference with the recommended return value from the Blockly.inject() call in the editor.js file and the UI appears to be functioning as expected. I would like to explore this further before implementing any further changes. Is it possible that using the workspace is "how it's always been done" and that is why it is still being used?

**Comment by MatzElectronics** [comment]
No - there is no good reason, other than it's always worked, and no one, myself included, looked it up or knew any different.

There is a lot of `Blockly.getMainWorkspace()` calls throughout the code base, including in many blocks.  

Unless there is a reason not to do this, I think we should attach the rendered workspace **back to the `Blockly`** object.  For example, we could extend Blockly with `setBlocklyPropWorkspace()` and `getBlocklyPropWorkspace()` prototypes.

**Comment by zfi** [comment]
Agreed, there are quite a few references to the getMainWorkspace() in the code. I am unsure what the distinction is inside Blockly core that makes any difference. I'll post that question over on the Blockly group.

The code in the feature-modules branch captures the return value from the Blockly.inject call in a module-local variable. The module exposes a getter method that other modules can use to obtain a reference to the WorkspaceSvg object. It has the added benefit in that the workspace methods all resolve correctly now. 

I think this will work for our purposes since there is only one call to Blockly.inject in the entire project and that is located in the editor.js module. I would like to avoid altering the Blockly core in any way if it can be avoided. There will be less baggage to deal with during subsequent code updates.

**Comment by MatzElectronics** [comment]
I think that it's only an issue if there are multiple instances/workspaces on a single web site or page.  We only ever use one instance, so that's why it's always worked.  Otherwise, from the little bit of googling I did on the topic, it will return whatever workspace is focused - roughly speaking.
